File: src/main/java/com/github/ccguyka/showupdates/producer/BasicUpdatesSource.java
package com.github.ccguyka.showupdates.producer;

import java.util.Collection;
import java.util.stream.Collectors;

import org.apache.maven.artifact.Artifact;
import org.apache.maven.artifact.versioning.ArtifactVersion;

import com.github.ccguyka.showupdates.objects.ArtifactUpdate;

class BasicUpdatesSource {

    protected ArtifactUpdate from(final Artifact artifact, final Collection<ArtifactVersion> artifactVersion) {
        if (artifactVersion == null || artifactVersion.isEmpty()) {
            return null;
        }
        return new ArtifactUpdate(artifact.getGroupId() + ":" + artifact.getArtifactId(), artifact.getVersion(),
                artifactVersion.stream().map(ArtifactVersion::toString).collect(Collectors.toList()));
    }
}

File: src/main/kotlin/com/github/ccguyka/showupdates/producer/BasicUpdatesSource.kt
package com.github.ccguyka.showupdates.producer

import com.github.ccguyka.showupdates.objects.ArtifactUpdate
import org.apache.maven.artifact.Artifact
import org.apache.maven.artifact.versioning.ArtifactVersion
import java.util.stream.Collectors

open class BasicUpdatesSource {

    protected fun from(artifact: Artifact, artifactVersion: Collection<ArtifactVersion>?): ArtifactUpdate? {
        return if (artifactVersion == null || artifactVersion.isEmpty()) {
            null
        } else ArtifactUpdate(artifact.groupId + ":" + artifact.artifactId, artifact.version,
                artifactVersion.stream().map { obj: ArtifactVersion -> obj.toString() }.collect(Collectors.toList()))
    }
}
